graph/patterns/subsets.md

```diff
@@ -0,0 +1,173 @@
+
+*The modeling collection subsets pattern is the modeling state associated to a collection that may include all instances, an included subset, an excluded subset, no instances, or any combinations of the preceding items.*
+
+## Problem
+
+A common pattern is to apply a policy or state to a collection of objects. With this, there also comes the question of how to model cases where we want to apply to 'all' or 'none' without having to special case these values within the collection set or introduce cross-property dependencies. Likewise, we'd like to model it in a way where it is easy to understand and interpret usage from just looking at the schema.
+
+An example is where you have a policy that you need to be able to apply to users in an organization. You might want to support the default **None**, enablement for **All**, or enablement for **Select** users where you only grant it to a few users.
+
+Existing patterns for this either have special-cased 'strings' or have tightly coupled dependencies between two independent properties. Neither is intuitive, both require reading documentation, and neither can be inferred from the schema or within client libraries.
+
+## Solution
+
+Have an abstract base class where all 'variants' of the subset are derived types from the base subset. For more information, see the [general subtyping guidance](https://github.com/microsoft/api-guidelines/blob/graph/graph/patterns/subtypes.md).
+
+The abstract base class should also hold an enum for all possible variants. The purpose of including this is to allow for easier ways to do query and filter operations on variants like 'all' and 'none' without relying on `isof` functions.
+
+**Base type**
+
+```xml
+    <ComplexType Name="membership" IsAbstract="true">
+      <Property Name="membershipKind" Type="graph.membershipKind"/>
+    </ComplexType>
+
+    <Enum Name="membershipKind">
+      <Member Name="all"/>
+      <Member Name="enumerated"/>
+      <Member Name="none"/>
+      <Member Name="unknownFutureValue"/>
+    </Enum>
+```
+
+**Derived types**
+
+```xml
+    <ComplexType Name="noMembership" BaseType="graph.membership"/>
+
+    <ComplexType Name="allMembership" BaseType="graph.membership"/>
+
+    <ComplexType Name="enumeratedMembership" BaseType = "graph.membership">
+      <Property Name="members" Type="Collection(Edm.String)"/>
+    </ComplexType>
+
+    <ComplexType Name="excludedMembership" BaseType="graph.membership">
+      <Property Name="members" Type="Collection(Edm.String)"/>
+    </ComplexType>
+```
+
+Be aware that the name values and types in the preceding examples are just examples and can be replaced with your scenario equivalent values. For example, types don't really need to be memberships. The collection doesn't have to be a collection at all; it can be singular and doesn't have to be a string.
+
+```xml
+    <ComplexType Name="<base-type>" IsAbstract="true">
+      <Property Name="<enum-name-for-type>" Type="graph.<enum-type>"/>
```

Comment:
I know this wasn't in the original doc but I think we should standardize the relationship between the name of the abstract base type and the name of the enumeration type.

Reply:
@garethj-msft Is there a specific change that you wanted here?